### PR TITLE
[java] data flow and docstrings improvements

### PR DESCRIPTION
**Pull request by iampopovich**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

- I replaced the if statement with a more readable switch case.
- I consolidated duplicate switch case flows as done in other cases where it was appropriate and didn't compromise code readability.
- I replaced manual array copying with System.arraycopy and added a check for null args. I believe explicitly replacing null with an empty list won't confuse a new developer reading the code. Additionally, the array copying operation is now explicit.
- I fixed minor typos in docstrings.
- I corrected the code formatting in EventFiringDecorator.java so that the indentations now have a consistent style.

To ensure the code's functionality, I ran CI [locally](https://github.com/iampopovich/selenium/actions/runs/5200268037)

### Motivation and Context

I wanted to simplify certain parts of the code to improve readability and replace manual implementations with built-in Java functions.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: java/src/org/openqa/selenium/support/events/EventFiringDecorator.java [hunk 1]
```diff
@@ -233,12 +233,10 @@ private void fireBeforeEvents(
...
     String methodName = createEventMethodName("before", method.getName());
 
-    int argsLength = args != null ? args.length : 0;
-    Object[] args2 = new Object[argsLength + 1];
+    args = args != null ? args : new Object[0];
+    Object[] args2 = new Object[args.length + 1];
     args2[0] = target.getOriginal();
-    for (int i = 0; i < argsLength; i++) {
-      args2[i + 1] = args[i];
-    }
```
Comment (diemol) [inline java/src/org/openqa/selenium/support/events/EventFiringDecorator.java:241]: Is this more readable? I am not sure how this improves the code, I get the impression that the current version is fine.

File: java/src/org/openqa/selenium/support/events/EventFiringDecorator.java [hunk 2]
```diff
@@ -247,17 +245,16 @@ private void fireBeforeEvents(
...
-        method.getReturnType() == Void.TYPE || method.getReturnType() == WebDriver.Timeouts.class;
-    int argsLength = args != null ? args.length : 0;
-    Object[] args2 = new Object[argsLength + 1 + (isVoid ? 0 : 1)];
+      method.getReturnType() == Void.TYPE || method.getReturnType() == WebDriver.Timeouts.class;
+    args = args != null ? args : new Object[0];
+    Object[] args2 = new Object[args.length + 1 + (isVoid ? 0 : 1)];
     args2[0] = target.getOriginal();
-    for (int i = 0; i < argsLength; i++) {
-      args2[i + 1] = args[i];
-    }
```
Comment (diemol) [inline java/src/org/openqa/selenium/support/events/EventFiringDecorator.java:260]: Same here.

File: java/src/org/openqa/selenium/support/events/EventFiringDecorator.java [hunk 3]
```diff
@@ -297,8 +294,8 @@ private void fireAfterEvents(
 
   private String createEventMethodName(String prefix, String originalMethodName) {
     return prefix
-        + originalMethodName.substring(0, 1).toUpperCase()
-        + originalMethodName.substring(1);
```
Comment (diemol) [inline java/src/org/openqa/selenium/support/events/EventFiringDecorator.java:301]: Please run the `format.sh` script to double check this changes. It is located under the `scripts` directory.

File: java/src/org/openqa/selenium/support/events/EventFiringDecorator.java [hunk 4]
```diff
@@ -157,7 +157,7 @@
  * but the decorator suppresses these exceptions), can't prevent execution of the decorated methods,
  * can't modify parameters and results of the methods.
  *
- * <p>Decorators that modify the behaviour of the underlying drivers should be implemented by
```
Comment (diemol) [inline java/src/org/openqa/selenium/support/events/EventFiringDecorator.java:160]: Most of the comments use British English, not that it matters, but I am not sure if this is a substantial change that needs to be done.

**Comment by iampopovich** [comment]
@diemol Thank you for the response 🙏 I will take the suggestions into account in the future. The proposed changes are minor, and if they are not significant, I will go ahead and close the pull request for now.

**Pull request closed without merging**